keras_nlp/models/whisper/whisper_preprocessor.py

```diff
@@ -0,0 +1,302 @@
+            start_value=bos_tokens,
+            end_value=self.tokenizer.eos_token_id,
+            pad_value=self.tokenizer.pad_token_id,
+            sequence_length=decoder_sequence_length,
+            return_padding_mask=True,
+        )
+
+        self.language = language
+        self.task = task
+        self.no_timestamps = no_timestamps
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "decoder_sequence_length": self.decoder_packer.sequence_length,
+                "language": self.language,
+                "task": self.task,
+                "no_timestamps": self.no_timestamps,
+            }
+        )
+        return config
+
+    def call(self, x, y=None, sample_weight=None):
+        if not (
+            isinstance(x, dict)
+            and ["encoder_audio", "decoder_text"] == list(x.keys())
+        ):
+            raise ValueError(
+                '`x` must be a dictionary, containing the keys `"encoder_audio"`'
+                f' and `"decoder_text"`. Received x={x}.'
+            )
+
+        encoder_audio = x["encoder_audio"]
+        decoder_text = x["decoder_text"]
+
+        encoder_audio = convert_inputs_to_list_of_tensor_segments(encoder_audio)
+        decoder_text = convert_inputs_to_list_of_tensor_segments(decoder_text)
+
+        if len(encoder_audio) > 1 or len(decoder_text) > 1:
+            raise ValueError(
+                '`WhisperPreprocessor` requires both `"encoder_audio"` and '
+                f'`"decoder_text"` to contain only one segment, but received '
+                f"{len(encoder_audio)} and {len(decoder_text)}, respectively."
+            )
+
+        encoder_features = self.audio_feature_extractor(encoder_audio[0])
+
+        decoder_inputs = self.tokenizer(decoder_text[0])
+        decoder_token_ids, decoder_padding_mask = self.decoder_packer(
+            decoder_inputs
+        )
+
+        x = {
+            "encoder_features": encoder_features,
+            "decoder_token_ids": decoder_token_ids,
+            "decoder_padding_mask": decoder_padding_mask,
+        }
+
+        return pack_x_y_sample_weight(x, y, sample_weight)
+
+    @classproperty
+    def audio_feature_extractor_cls(cls):
+        return WhisperAudioFeatureExtractor
+
+    @classproperty
+    def tokenizer_cls(cls):
+        return WhisperTokenizer
+
+    @classproperty
+    def presets(cls):
+        return copy.deepcopy(backbone_presets)
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        language=None,
+        task=None,
+        no_timestamps=True,
+        **kwargs,
+    ):
+        """Instantiate `WhisperPreprocessor` from preset architecture.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+            language: string, language tokens. Should only be passed if your
+                tokenizer is multilingual. Must be one of "{{language_tokens}}".
```

Comment:
I wouldn't go so far as listing all the language tokens here, this is going to cause us trouble with maintenance down the line, especially we we try to move back to shared base class `from_preset`. Can we just document the form of the language tokens here, and throw an error in the body of this method if it's not supported?